x-pack/plugins/ml/public/data_frame/components/source_index_preview/use_source_index_data.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { SearchResponse } from 'elasticsearch';
+
+import { ml } from '../../../services/ml_api_service';
+
+import { SimpleQuery } from '../../common';
+import { IndexPatternContextValue } from '../../common/index_pattern_context';
+import { EsDoc, EsFieldName, getDefaultSelectableFields } from './common';
+
+const SEARCH_SIZE = 1000;
+
+export enum SOURCE_INDEX_STATUS {
+  UNUSED,
+  LOADING,
+  LOADED,
+  ERROR,
+}
+
+export interface UseSourceIndexDataReturnType {
+  errorMessage: string;
+  status: SOURCE_INDEX_STATUS;
+  tableItems: EsDoc[];
+}
+
+export const useSourceIndexData = (
+  indexPattern: IndexPatternContextValue,
+  query: SimpleQuery,
+  selectedFields: EsFieldName[],
+  setSelectedFields: React.Dispatch<React.SetStateAction<EsFieldName[]>>
+): UseSourceIndexDataReturnType => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [status, setStatus] = useState(SOURCE_INDEX_STATUS.UNUSED);
+  const [tableItems, setTableItems] = useState([] as EsDoc[]);
+
+  if (indexPattern !== null) {
+    const getSourceIndexData = async function() {
+      setErrorMessage('');
+      setStatus(SOURCE_INDEX_STATUS.LOADING);
+
+      try {
+        const resp: SearchResponse<any> = await ml.esSearch({
+          index: indexPattern.title,
+          rest_total_hits_as_int: true,
```

Comment:
Do you need to specify `rest_total_hits_as_int` as you don't actually need the hit count?

Reply:
We're not using it so I removed it in 70c149c.